### PR TITLE
[stable/coredns] add optional prometheus service monitor to coredns

### PR DESCRIPTION
**Pull request by rmb938**
#### What this PR does / why we need it:

add optional prometheus service monitor to corends

#### Special notes for your reviewer:

other charts like the nginx-ingress has this as a feature so I'm adding it to corends as well

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by rmb938** [comment]
Hi @shashidharatd could you take a look at this?

**Comment by rmb938** [comment]
@shashidharatd could I get some input on this?

**Comment by mrueg** [comment]
@gianrubio any thoughts on integrating it here vs. https://github.com/helm/charts/tree/master/stable/prometheus-operator/templates/exporters/core-dns

**Comment by rmb938** [comment]
@mrueg I don't want to have to install the prometheus-operator chart to get this feature.

**Comment by rmb938** [comment]
@shashidharatd Can you look at this again, I resynced it with master so there are no merge conflicts now.

File: stable/coredns/templates/service-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    component: metrics
+  annotations:
+{{ toYaml .Values.service.annotations | indent 4 }}
+spec:
+  selector:
+    release: {{ .Release.Name | quote }}
+    {{- if .Values.isClusterService }}
+    k8s-app: {{ .Chart.Name | quote }}
+    {{- end }}
+    app: {{ template "coredns.name" . }}
```
Comment (mrueg) [inline stable/coredns/templates/service-metrics.yaml]: this label needs to be: `app.kubernetes.io/name`

File: stable/coredns/templates/service-metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    kubernetes.io/cluster-service: "true"
+    kubernetes.io/name: "CoreDNS"
+    {{- end }}
+    app: {{ template "coredns.name" . }}
+    component: metrics
+  annotations:
+{{ toYaml .Values.service.annotations | indent 4 }}
+spec:
+  selector:
+    release: {{ .Release.Name | quote }}
```
Comment (mrueg) [inline stable/coredns/templates/service-metrics.yaml]: this label needs to be: `app.kubernetes.io/instance`

File: stable/coredns/templates/service-metrics.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    {{- if .Values.isClusterService }}
+    k8s-app: {{ .Chart.Name | quote }}
+    kubernetes.io/cluster-service: "true"
+    kubernetes.io/name: "CoreDNS"
+    {{- end }}
+    app: {{ template "coredns.name" . }}
+    component: metrics
```
Comment (mrueg) [inline stable/coredns/templates/service-metrics.yaml]: this should be `app.kubernetes.io/component`

File: stable/coredns/templates/service-metrics.yaml (hunk 1 above)
Comment (rmb938) [reply stable/coredns/templates/service-metrics.yaml]: fixed

File: stable/coredns/templates/service-metrics.yaml (hunk 2 above)
Comment (rmb938) [reply stable/coredns/templates/service-metrics.yaml]: fixed

File: stable/coredns/templates/service-metrics.yaml (hunk 3 above)
Comment (rmb938) [reply stable/coredns/templates/service-metrics.yaml]: fixed

**Review by mrueg — APPROVED** [review]

**Comment by mrueg** [comment]
@shashidharatd friendly ping on this one

**Comment by shashidharatd** [comment]
@mrueg, I am really sorry for not noticing this sooner. Will get to it ASAP.

**Comment by shashidharatd** [comment]
This LGTM. Thank you for bearing with the delays.
/lgtm

**Pull request merged**